dotnet/samples/GettingStartedWithAgents/AzureAIAgent/Step08_AzureAIAgent_Declarative.cs

```diff
@@ -0,0 +1,135 @@
+using Microsoft.SemanticKernel.ChatCompletion;
+using Plugins;
+
+namespace GettingStarted.AzureAgents;
+
+/// <summary>
+/// This example demonstrates how to declaratively create instances of <see cref="AzureAIAgent"/>.
+/// </summary>
+public class Step08_AzureAIAgent_Declarative : BaseAzureAgentTest
+{
+    public Step08_AzureAIAgent_Declarative(ITestOutputHelper output) : base(output)
+    {
+        var builder = Kernel.CreateBuilder();
+        builder.Services.AddSingleton<AIProjectClient>(this.Client);
+        this._kernel = builder.Build();
+    }
+
+    [Fact]
+    public async Task AzureAIAgentWithCodeInterpreterAsync()
+    {
+        var text =
+            """
+            type: azureai_agent
+            name: AzureAIAgent
+            description: AzureAIAgent Description
+            instructions: AzureAIAgent Instructions
+            model:
+              id: gpt-4o-mini
+            tools:
+                - name: tool1
+                  type: code_interpreter
+            """;
+        AzureAIAgentFactory factory = new();
+
+        var agent = await factory.CreateAgentFromYamlAsync(this._kernel, text) as AzureAIAgent;
+
+        await InvokeAgentAsync(agent, "Use code to determine the values in the Fibonacci sequence that that are less then the value of 101?");
+    }
+
+    [Fact(Skip = "Support for tool configuration will be added in a later PR")]
+    public async Task AzureAIAgentWithOpenApiAsync()
+    {
+        var text =
+            """
+            type: azureai_agent
+            name: AzureAIAgent
+            description: AzureAIAgent Description
+            instructions: AzureAIAgent Instructions
+            model:
+              id: gpt-4o-mini
+            tools:
+              - type: openapi
+                name: RestCountriesAPI
+                description: Web API version 3.1 for managing country items, based on previous implementations from restcountries.eu and restcountries.com.
+                schema: '{"openapi":"3.1.0","info":{"title":"RestCountries.NET API","description":"Web API version 3.1 for managing country items, based on previous implementations from restcountries.eu and restcountries.com.","version":"v3.1"},"servers":[{"url":"https://restcountries.net"}],"auth":[],"paths":{"/v3.1/currency":{"get":{"description":"Search by currency.","operationId":"LookupCountryByCurrency","parameters":[{"name":"currency","in":"query","description":"The currency to search for.","required":true,"schema":{"type":"string"}}],"responses":{"200":{"description":"Success","content":{"text/plain":{"schema":{"type":"string"}}}}}}}},"components":{"schemes":{}}}'
+            """;
+        AzureAIAgentFactory factory = new();
+
+        var agent = await factory.CreateAgentFromYamlAsync(this._kernel, text) as AzureAIAgent;
+
+        await InvokeAgentAsync(agent, "Use code to determine the values in the Fibonacci sequence that that are less then the value of 101?");
+    }
```

Comment:
IF this test is not for manual verification and is not supported, I would remove it and keep track of the removal commit for later addition in the aforementioned PR.
 
```suggestion
```